src/control/cmd/daos_server/auto.go

```diff
@@ -60,17 +60,48 @@ func getLocalFabric(ctx context.Context, log logging.Logger) (*control.HostFabri
 	return hf, nil
 }
 
-type getStorageFn func(context.Context, logging.Logger) (*control.HostStorage, error)
+type getStorageFn func(context.Context, logging.Logger, bool) (*control.HostStorage, error)
 
-func getLocalStorage(ctx context.Context, log logging.Logger) (*control.HostStorage, error) {
+func getLocalStorage(ctx context.Context, log logging.Logger, skipPrep bool) (*control.HostStorage, error) {
 	svc := server.NewStorageControlService(log, config.DefaultServer().Engines).
 		WithVMDEnabled() // use vmd if present
 
+	var nc *nvmeCmd
+	if !skipPrep {
+		nc = &nvmeCmd{}
+		nc.Logger = log
+		nc.IgnoreConfig = true
+		if err := nc.init(); err != nil {
+			return nil, errors.Wrap(err, "could not init nvme cmd")
+		}
+
+		req := storage.BdevPrepareRequest{}
+		if err := prepareNVMe(req, nc, svc.NvmePrepare); err != nil {
+			return nil, errors.Wrap(err, "nvme prep before fetching local storage failed, "+
+				"try cmd again with --skip-prep after performing a manual nvme prepare")
+		}
+	}
+
 	nvmeResp, err := svc.NvmeScan(storage.BdevScanRequest{})
 	if err != nil {
 		return nil, errors.Wrapf(err, "nvme scan")
 	}
 
+	if !skipPrep {
+		// TODO SPDK-2926: If VMD is enabled and PCI_ALLOWED list is set to a subset of VMD
+		//                 controllers (as specified in the server config file) then the
+		//                 backing devices of the unselected VMD controllers will be bound
+		//                 to no driver and therefore inaccessible from both OS and SPDK.
+		//                 Workaround is to run nvme scan --ignore-config to reset driver
+		//                 bindings.
+
+		req := storage.BdevPrepareRequest{Reset_: true}
+		if err := resetNVMe(req, nc, svc.NvmePrepare); err != nil {
+			return nil, errors.Wrap(err, "nvme reset after fetching local storage failed, "+
+				"try cmd again with --skip-prep after performing a manual nvme reset")
```

Comment:
Might be nice to give users the details on how to do the manual reset in the error message.